File: src/leetcode/2026/02/Solution645.cpp
#include <vector>
using namespace std;

class Solution645 {
public:
    vector<int> findErrorNums(vector<int>& nums) {
        int n = nums.size();
        vector<int> count(n + 1, 0);
        for (int num : nums) {
            count[num]++;
        }
        vector<int> result(2);
        for (int i = 1; i <= n; i++) {
            if (count[i] == 2) {
                result[0] = i; // Duplicate number
            } else if (count[i] == 0) {
                result[1] = i; // Missing number
            }
        }
        return result;
    }
};

int main() {
    Solution645 solution;

    vector<int> nums = {1, 2, 2, 4};
    vector<int> result = solution.findErrorNums(nums);
    // Output: [2, 3]
    printf("Duplicate: %d, Missing: %d\n", result[0], result[1]);

    vector<int> nums2 = {1, 1};
    vector<int> result2 = solution.findErrorNums(nums2);
    // Output: [1, 2]
    printf("Duplicate: %d, Missing: %d\n", result2[0], result2[1]);

    return 0;
}

File: src/leetcode/2026/02/Solution645.java
public class Solution645 {
    public int[] findErrorNums(int[] nums) {
        int n = nums.length;
        int[] count = new int[n + 1];
        for (int num : nums) {
            count[num]++;
        }
        int duplicate = -1, missing = -1;
        for (int i = 1; i <= n; i++) {
            if (count[i] == 2) {
                duplicate = i;
            } else if (count[i] == 0) {
                missing = i;
            }
        }
        return new int[] { duplicate, missing };
    }

    public static void main(String[] args) {
        Solution645 solution = new Solution645();

        int[] nums1 = { 1, 2, 2, 4 };
        int[] result1 = solution.findErrorNums(nums1);
        System.out.println("Duplicate: " + result1[0] + ", Missing: " + result1[1]); // Should print Duplicate: 2,
                                                                                     // Missing: 3

        int[] nums2 = { 1, 1 };
        int[] result2 = solution.findErrorNums(nums2);
        System.out.println("Duplicate: " + result2[0] + ", Missing: " + result2[1]); // Should print Duplicate: 1,
                                                                                     // Missing: 2
    }
}
